packages/agents/src/react.tsx

```diff
@@ -39,6 +39,7 @@ export type UseAgentOptions<State = unknown> = Omit<
   name?: string;
   /** Called when the Agent's state is updated */
   onStateUpdate?: (state: State, source: "server" | "client") => void;
+  onMcpUpdate?: (mcpServers: MCPServerState) => void;
```

Comment:
Not sure if we want to expose an `addMcpServer()` function as a native capability of `useAgent`, but I decided to just provide a hook into the reading MCP server state. We could always add that functionality later.

Reply:
yeah we probably don't want to explicitly add addMcpServer here, and it's trivial to expose it with rpc anyway